[GPU Process] [FormControls] Share the ApplePayButton drawing among all Cocoa platforms

shallawa opened the pull request:
#### 9bc94b5f4fb6121f2f0fb01688a2f596aa76324f
<pre>
[GPU Process] [FormControls] Share the ApplePayButton drawing among all Cocoa platforms
<a href="https://bugs.webkit.org/show_bug.cgi?id=251708">https://bugs.webkit.org/show_bug.cgi?id=251708</a>
rdar://105020333

Reviewed by Aditya Keerthi.

Remove RenderThemeCocoa::paintApplePayButton() and create a ControlPart for the
ApplePayButton on iOS so it can be drawn by ApplePayButtonCocoa::draw().

Add a ControlFactory for iOS and add stub creators for the platform controls.
These creators will be filled in the future to replace the existing code in
RenderThemeIOS with iOS platform controls.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay/ApplePayButtonSystemImage.h: Removed.
* Source/WebCore/Modules/applepay/ApplePayButtonSystemImage.mm: Removed.
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/platform/graphics/SystemImage.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm:
(WebCore::toPKPaymentButtonType):
(WebCore::toPKPaymentButtonStyle):
(WebCore::ApplePayButtonCocoa::draw):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.h:
* Source/WebCore/platform/graphics/controls/ControlFactory.cpp:
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.h: Added.
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.mm: Added.
(WebCore::ControlFactory::createControlFactory):
(WebCore::ControlFactoryIOS::createPlatformButton):
(WebCore::ControlFactoryIOS::createPlatformColorWell):
(WebCore::ControlFactoryIOS::createPlatformInnerSpinButton):
(WebCore::ControlFactoryIOS::createPlatformMenuList):
(WebCore::ControlFactoryIOS::createPlatformMenuListButton):
(WebCore::ControlFactoryIOS::createPlatformMeter):
(WebCore::ControlFactoryIOS::createPlatformProgressBar):
(WebCore::ControlFactoryIOS::createPlatformSearchField):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldCancelButton):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldResults):
(WebCore::ControlFactoryIOS::createPlatformSliderThumb):
(WebCore::ControlFactoryIOS::createPlatformSliderTrack):
(WebCore::ControlFactoryIOS::createPlatformTextArea):
(WebCore::ControlFactoryIOS::createPlatformTextField):
(WebCore::ControlFactoryIOS::createPlatformToggleButton):
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.mm:
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.mm:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/RenderThemeCocoa.h:
* Source/WebCore/rendering/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintApplePayButton): Deleted.
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canCreateControlPartForRenderer const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder&lt;SystemImage&gt;::encode):
(IPC::ArgumentCoder&lt;SystemImage&gt;::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/259878@main">https://commits.webkit.org/259878@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c1c79ea16b8299d3fb0d6b20cb769ad667d4faac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106284 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15338 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39123 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115477 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/175580 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110194 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16780 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6554 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98500 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115162 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112048 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12765 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95745 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40319 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94640 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27388 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/82009 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8578 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28740 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9088 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5312 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14701 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48286 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6826 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10624 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm:
```diff
@@ -48,14 +99,22 @@
+        CGRectMake(logicalRect.x(), -logicalRect.maxY(), logicalRect.width(), logicalRect.height()),
+        1.0,
+        largestCornerRadius,
+        toPKPaymentButtonType(owningApplePayButtonPart().buttonType()),
```
Nit: Consider storing `owningApplePayButtonPart()` in a local variable, since it's used 3 times.

shallawa committed:
[GPU Process] [FormControls] Share the ApplePayButton drawing among all Cocoa platforms
https://bugs.webkit.org/show_bug.cgi?id=251708
rdar://105020333

Reviewed by Aditya Keerthi.

Remove RenderThemeCocoa::paintApplePayButton() and create a ControlPart for the
ApplePayButton on iOS so it can be drawn by ApplePayButtonCocoa::draw().

Add a ControlFactory for iOS and add stub creators for the platform controls.
These creators will be filled in the future to replace the existing code in
RenderThemeIOS with iOS platform controls.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/applepay/ApplePayButtonSystemImage.h: Removed.
* Source/WebCore/Modules/applepay/ApplePayButtonSystemImage.mm: Removed.
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/platform/graphics/SystemImage.h:
* Source/WebCore/platform/graphics/cocoa/controls/ApplePayButtonCocoa.mm:
(WebCore::toPKPaymentButtonType):
(WebCore::toPKPaymentButtonStyle):
(WebCore::ApplePayButtonCocoa::draw):
* Source/WebCore/platform/graphics/controls/ApplePayButtonPart.h:
* Source/WebCore/platform/graphics/controls/ControlFactory.cpp:
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.h: Added.
* Source/WebCore/platform/graphics/ios/controls/ControlFactoryIOS.mm: Added.
(WebCore::ControlFactory::createControlFactory):
(WebCore::ControlFactoryIOS::createPlatformButton):
(WebCore::ControlFactoryIOS::createPlatformColorWell):
(WebCore::ControlFactoryIOS::createPlatformInnerSpinButton):
(WebCore::ControlFactoryIOS::createPlatformMenuList):
(WebCore::ControlFactoryIOS::createPlatformMenuListButton):
(WebCore::ControlFactoryIOS::createPlatformMeter):
(WebCore::ControlFactoryIOS::createPlatformProgressBar):
(WebCore::ControlFactoryIOS::createPlatformSearchField):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldCancelButton):
(WebCore::ControlFactoryIOS::createPlatformSearchFieldResults):
(WebCore::ControlFactoryIOS::createPlatformSliderThumb):
(WebCore::ControlFactoryIOS::createPlatformSliderTrack):
(WebCore::ControlFactoryIOS::createPlatformTextArea):
(WebCore::ControlFactoryIOS::createPlatformTextField):
(WebCore::ControlFactoryIOS::createPlatformToggleButton):
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlFactoryMac.mm:
* Source/WebCore/platform/graphics/mac/controls/ToggleButtonMac.mm:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/RenderThemeCocoa.h:
* Source/WebCore/rendering/RenderThemeCocoa.mm:
(WebCore::RenderThemeCocoa::paintApplePayButton): Deleted.
* Source/WebCore/rendering/RenderThemeIOS.h:
* Source/WebCore/rendering/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::canCreateControlPartForRenderer const):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.cpp:
(IPC::ArgumentCoder<SystemImage>::encode):
(IPC::ArgumentCoder<SystemImage>::decode):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/259878@main
